client/alloc_runner.go

```diff
@@ -257,6 +269,7 @@ func (r *AllocRunner) setTaskStatus(taskName, status, desc string) {
 		Description: desc,
 	}
 	r.taskStatusLock.Unlock()
+	r.SaveState(taskName)
```

Comment:
It's not enough to save the Task Runner state, since the Alloc state could have changed too, once the Task state has changed